pkg/execution/executor/executor.go

```diff
@@ -1487,6 +1508,28 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 			return nil, err
 		}
 
+		if e.allowStepMetadata.Enabled(ctx, instance.Metadata().ID.Tenant.AccountID) {
+			for _, opcode := range resp.Generator {
+				for _, metadata := range opcode.Metadata {
+					if err := metadata.Validate(); err != nil {
+						l.Warn("invalid metadata in driver response", "error", err)
```

Comment:
A note is that if we could add something like `l.With("step_metadata", true)` to all logs around this feature, it'd be a nice easy way to spot issues/warnings as we roll out.